app/core/Engine.ts

```diff
@@ -509,9 +541,17 @@ class Engine {
     });
 
     const tokensController = new TokensController({
-      // TODO: The tokens controller currently does not support internalAccounts. This is done to match the behavior of the previous tokens controller subscription.
       onPreferencesStateChange: (listener) =>
-        preferencesController.subscribe(listener),
+        this.controllerMessenger.subscribe(
+          `${preferencesController.name}:stateChange`,
+          (preferencesControllerState: PreferencesState) => {
+            //@ts-expect-error assets-controllers need to be bump to v^26
+            //assets-controllers preferences controller version is misaligned with the app repo (v8 != v4)
+            listener({
+              ...preferencesControllerState,
+            });
+          },
```

Comment:
Nit: I don't think we need to shallow clone the state here either. Something like this might be simpler:

```suggestion
          //@ts-expect-error assets-controllers need to be bump to v^26
          //assets-controllers preferences controller version is misaligned with the app repo (v8 != v4)
          listener,
```